src/WebJobs.Script/Rpc/LanguageWorkerChannelManager.cs

```diff
@@ -0,0 +1,182 @@
+        private readonly IScriptEventManager _eventManager = null;
+        private readonly ILoggerFactory _loggerFactory = null;
+        private readonly IWorkerProcessFactory _processFactory;
+        private readonly IProcessRegistry _processRegistry;
+        private readonly IRpcServer _rpcServer = null;
+        private readonly IDisposable _rpcChannelReadySubscriptions;
+        private IDictionary<string, ILanguageWorkerChannel> _webhostChannels = new Dictionary<string, ILanguageWorkerChannel>();
+
+        public LanguageWorkerChannelManager(IScriptEventManager eventManager, IRpcServer rpcServer, ILoggerFactory loggerFactory, IOptions<LanguageWorkerOptions> languageWorkerOptions, IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions)
+        {
+            _rpcServer = rpcServer;
+            _eventManager = eventManager;
+            _loggerFactory = loggerFactory;
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryLanguageWorkerChannelManager);
+            _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
+            _applicationHostOptions = applicationHostOptions;
+            _processFactory = new DefaultWorkerProcessFactory();
+            try
+            {
+                _processRegistry = ProcessRegistryFactory.Create();
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning(e, "Unable to create process registry");
+            }
+            _rpcChannelReadySubscriptions = _eventManager.OfType<RpcChannelReadyEvent>()
+               .Subscribe(AddOrUpdateWebhostChannels);
+        }
+
+        public ILanguageWorkerChannel CreateLanguageWorkerChannel(string workerId, string scriptRootPath, string language, IObservable<FunctionRegistrationContext> functionRegistrations, IMetricsLogger metricsLogger, bool isWebhostChannel, int attemptCount)
+        {
+            var languageWorkerConfig = _workerConfigs.Where(c => c.Language.Equals(language, StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+            if (languageWorkerConfig == null)
+            {
+                throw new InvalidOperationException($"WorkerCofig for runtime: {language} not found");
+            }
+            return new LanguageWorkerChannel(
+                         workerId,
+                         scriptRootPath,
+                         _eventManager,
+                         functionRegistrations,
+                         _processFactory,
+                         _processRegistry,
+                         languageWorkerConfig,
+                         _rpcServer.Uri,
+                         _loggerFactory,
+                         metricsLogger,
+                         isWebhostChannel,
+                         attemptCount);
+        }
+
+        public async Task InitializeChannelAsync(string language)
+        {
+            _logger?.LogDebug("Initializing LaguageWorkerChannel at Webhost level");
```

Comment:
Need to remove all this WebHost logging/naming. Consider - this host can be run outside of Azure Functions completely, in cases where there is no WebHost at all. E.g. see the console sample [here](https://github.com/Azure/azure-functions-host/blob/2ef5d7223e5d2e2c44ca54e3becfa840af9303f5/src/WebJobs.Script.Host/Program.cs#L34)

Reply:
Also fix spelling in log message